toolchain/check/testdata/builtins/type/and.carbon

```diff
@@ -0,0 +1,101 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/builtins/type/and.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/builtins/type/and.carbon
+
+// --- combine_facets.carbon
+
+library "[[@TEST_NAME]]";
+
+fn TypeAnd(a: type, b: type) -> type = "type.and";
+
+interface I {}
+interface J {}
+
+fn TakeIJ(T:! TypeAnd(I, J)) {}
```

Comment:
Is this generally tested well enough that you don't care to see the IR in this case?

Reply:
Yeah, I think the diagnostics (and lack thereof) for these tests covered everything I wanted to test here, and we have lots of tests for `&` uses of elsewhere.